packages/amplify-category-geo/src/service-walkthroughs/geofenceCollectionWalkthrough.ts

```diff
@@ -0,0 +1,210 @@
+            Object.keys(crudPermissionsMap),
+            { returnSize: 'many', initial: byValues(defaults) }
+        );
+
+        return selectedCrudPermissions;
+    };
+
+    const selectedGroupPermissions: Record<string, string[]> = {};
+
+    for (const selectedUserPoolGroup of selectedUserPoolGroupsList) {
+        let defaults: string[] = [];
+
+        if (parameters.groupPermissions) {
+            defaults = parameters.groupPermissions[selectedUserPoolGroup] || [];
+        }
+
+        const selectedCrudPermissions = await groupCrudPermissionsFlow(selectedUserPoolGroup, defaults);
+
+        selectedGroupPermissions[selectedUserPoolGroup] = selectedCrudPermissions;
+    }
+
+    parameters.groupPermissions = selectedGroupPermissions;
+    return parameters;
+};
+
+/**
+ * Starting point for CLI walkthrough that updates an existing geofence collection resource
+ * @param context The Amplify Context object
+ * @param parameters The configurations of the geofence collection resource
+ * @param resourceToUpdate Name of the geofence collection resource to update
+ */
+export const updateGeofenceCollectionWalkthrough = async (
+    context: $TSContext,
+    parameters: Partial<GeofenceCollectionParameters>,
+    resourceToUpdate?: string
+): Promise<Partial<GeofenceCollectionParameters>> => {
+    const collectionResources = ((await context.amplify.getResourceStatus()).allResources as any[])
+    .filter(resource => resource.service === ServiceName.GeofenceCollection)
+
+    if (collectionResources.length === 0) {
+        printer.error(`No ${geofencingServiceFriendlyName} resource to update. Use "amplify add geo" to create a new ${geofencingServiceFriendlyName}.`);
+        return parameters;
+    }
+
+    const collectionResourceNames = collectionResources.map(resource => resource.resourceName);
+
+    if (resourceToUpdate) {
+        if (!collectionResourceNames.includes(resourceToUpdate)) {
+            printer.error(`No ${geofencingServiceFriendlyName} named ${resourceToUpdate} exists in the project.`);
+            return parameters;
+        }
+    }
+    else {
+        resourceToUpdate = await prompter.pick<'one', string>(`Select the ${geofencingServiceFriendlyName} you want to update`, collectionResourceNames);
+    }
+
+    parameters.name = resourceToUpdate;
+    parameters = merge(parameters, await getCurrentGeofenceCollectionParameters(resourceToUpdate));
+
+    // overwrite the parameters based on user input
+
+    parameters.groupPermissions = (await geofenceCollectionAccessWalkthrough(context, parameters)).groupPermissions;
+
+    const otherCollectionResources = collectionResourceNames.filter(collectionResourceName => collectionResourceName != resourceToUpdate);
+    // if this is the only geofence collection, default cannot be removed
+    if (otherCollectionResources.length > 0) {
+        const isDefault = await prompter.yesOrNo(defaultResourceQuestion(ServiceName.GeofenceCollection), true);
```

Comment:
shouldn't the initial value be set to whether the resource is currently the default or not?